### PR TITLE
feat(node): add rate limiting to RPC service (also backoff for metrics and evm)

### PR DESCRIPTION
**Pull request by dirvine**
* Implements per-IP and global rate limiting for RPC endpoints
* Adds resource exhaustion error handling
* Removes unused metrics buffer fields
* Fixes replication manager concurrency control

### Description

Please provide a brief description of the changes made in this pull request. Highlight the purpose of the changes and the problem they address.

### Related Issue

Fixes #<issue_number> (if applicable).

### Type of Change

Please mark the types of changes made in this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

Please ensure all of the following tasks have been completed:

- [ ] I have read the [contributing guidelines](CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have followed the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines for commit messages.
- [ ] I have verified my commit messages with [commitlint](https://commitlint.js.org/#/).

### CODE REVIEW

File: sn_node/src/replication.rs [hunk 1]
```diff
@@ -136,3 +138,32 @@ impl Node {
         });
     }
 }
+
+pub struct ReplicationManager {
+    max_concurrent_replications: usize,
+    active_replications: Arc<AtomicUsize>,
+}
+
+impl ReplicationManager {
```
Comment (maqi) [inline sn_node/src/replication.rs:147]: currently, there is `ReplicationFetcher` which limits the `concurrent fetch` to be `MAX_PARALLEL_FETCH` (20).

This is somehow duplicated ?

File: sn_node/src/bin/safenode/rpc_service.rs [hunk 2]
```diff
@@ -49,6 +117,9 @@ impl SafeNode for SafeNodeRpcService {
         &self,
         request: Request<NodeInfoRequest>,
     ) -> Result<Response<NodeInfoResponse>, Status> {
+        // Check rate limit before processing request
+        self.rate_limiter.check_rate_limit(request.remote_addr()).await?;
```
Comment (maqi) [inline sn_node/src/bin/safenode/rpc_service.rs:121]: in case of error (i.e. no resource available, shall the task be put into a pool and wait for a retry?)

File: sn_node/src/replication.rs [hunk 3]
```diff
@@ -136,3 +138,32 @@ impl Node {
         });
     }
 }
+
+pub struct ReplicationManager {
```
Comment (b-zee) [inline sn_node/src/replication.rs:142]: This isn't used anywhere?

File: sn_node/src/replication.rs (hunk 1 above)
Comment (b-zee) [reply sn_node/src/replication.rs:147]: Looks like dead code, it's not used anywhere. No clue why this is in this PR.

File: sn_node/src/metrics.rs [hunk 4]
```diff
@@ -43,6 +44,11 @@ pub(crate) struct NodeMetricsRecorder {
     // to track the uptime of the node.
     pub(crate) started_instant: Instant,
     pub(crate) uptime: Gauge,
+
+    // Add sampling rate control
+    sampling_interval: Duration,
```
Comment (b-zee) [inline sn_node/src/metrics.rs:49]: These properties aren't used anywhere? Why is this added?

File: sn_node/src/bin/safenode/rpc_service.rs [hunk 5]
```diff
@@ -29,6 +29,73 @@ use tokio::sync::mpsc::{self, Sender};
...
+use std::num::NonZeroU32;
+use governor::{
+    clock::DefaultClock,
+    middleware::NoOpMiddleware,
+    state::{InMemoryState, NotKeyed},
+    Quota, RateLimiter as Governor,
+};
+
+/// RateLimiter to prevent abuse of the RPC service
+struct RateLimiter {
```
Comment (b-zee) [inline sn_node/src/bin/safenode/rpc_service.rs:42]: RPC service isn't meant to be public, why is this a concern?
